File: clients/python/rag_client.py
import requests
from typing import Optional


class RAGClient:
    def __init__(self, base_url: str, api_key: Optional[str] = None, timeout: int = 10):
        self.base_url = base_url.rstrip("/")
        self.api_key = api_key
        self.timeout = timeout

    def ask(self, query: str) -> dict:
        headers = {"Content-Type": "application/json"}
        if self.api_key:
            headers["x-api-key"] = self.api_key
        r = requests.post(
            f"{self.base_url}/ask",
            json={"query": query},
            headers=headers,
            timeout=self.timeout,
        )
        r.raise_for_status()
        return r.json()

File: clients/js/ragClient.js
export class RAGClient {
  constructor({ baseUrl, apiKey, timeoutMs = 10000 }) {
    this.baseUrl = baseUrl.replace(/\/$/, "");
    this.apiKey = apiKey;
    this.timeoutMs = timeoutMs;
  }
  async ask(query) {
    const controller = new AbortController();
    const to = setTimeout(() => controller.abort(), this.timeoutMs);
    const headers = { "Content-Type": "application/json" };
    if (this.apiKey) headers["x-api-key"] = this.apiKey;
    const res = await fetch(`${this.baseUrl}/ask`, {
      method: "POST",
      headers,
      body: JSON.stringify({ query }),
      signal: controller.signal,
    });
    clearTimeout(to);
    if (!res.ok) {
      const txt = await res.text();
      throw new Error(`HTTP ${res.status}: ${txt}`);
    }
    return res.json();
  }
}
